addons/mail/static/src/models/link_preview_list_view.js

```diff
@@ -0,0 +1,102 @@
+/** @odoo-module **/
+
+import { registerModel } from '@mail/model/model_core';
+import { many, one } from '@mail/model/model_field';
+import { clear, insertAndReplace, replace } from '@mail/model/model_field_command';
+
+registerModel({
+    name: 'LinkPreviewListView',
+    identifyingFields: [['composerView', 'messageView']],
+    recordMethods: {
+        /**
+         * @private
+         * @returns {FieldCommand}
+         */
+        _computeLinkPreviews() {
+            if (this.composerView) {
+                return replace(this.composerView.composer.linkPreviews);
+            }
+            if (this.messageView) {
+                return replace(this.messageView.message.linkPreviews);
+            }
+            return clear();
+        },
```

Comment:
This is bad, it will kill the composer/message relation, so you can never guess where is a link preview come from. And I need to know this information to determine `isDeletable` and inside `remove`.

It also doesn't make sens, LinkPreviews are data belonging to a composer or a message, storing them on `LinkPreviewListView` is wierd to me.

Reply:
Ok, ignore this comment. I forgot `linkPreview` is pure logic model while `LinkPreviewListView` is view model. So it's okay with relations on `composer` and `message`.